File: basic/src/main/java/com/yuanshenbin/basic/delegate/IDelegate.java
package com.yuanshenbin.basic.delegate;

import android.content.Context;

import com.yuanshenbin.basic.adapter.CommonAdapter;
import com.yuanshenbin.basic.state.StateLayoutManager;
import com.yuanshenbin.basic.util.NetworkUtils;
import com.yuanshenbin.network.model.ResponseModel;

import java.io.Serializable;

/**
 * author : yuanshenbin
 * time   : 2018/10/24
 * desc   :
 */

abstract class IDelegate implements Serializable {

    public void onShowToast(Context context, String str) {

    }


    public void onStateLayout(ResponseModel model, StateLayoutManager layoutManager) {
        if (layoutManager != null && model != null) {
            switch (model.getState()) {
                case 开始:
                    layoutManager.showLoading();
                    break;
                case 成功:
                    layoutManager.showContent();
                    break;
                case 结束:
                    break;
                case 失败:
                    if (NetworkUtils.isNetworkConnected(layoutManager.getContext())) {
                        layoutManager.getNetworkState().showNetwork("网络似乎开小差了", "重试");
                    } else {
                        layoutManager.showNetwork();
                    }
                    break;
                default:
            }
        }
    }

    public boolean onListDataEmpty(boolean pullAndPush, Object result) {


        return false;
    }

    public void handleListData(CommonAdapter adapter, Object result, int loading, int size, CharSequence emptyMsg){}


    public void onTitle(String title) {

    }

    public void onStart() {

    }

    public void onStop() {
    }

    public void onResume() {
    }

    public void onPause() {

    }

    public void onReload() {

    }

}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/delegate/IDelegate.kt
package com.yuanshenbin.basic.delegate

import android.content.Context
import com.yuanshenbin.basic.adapter.CommonAdapter
import com.yuanshenbin.basic.state.StateLayoutManager
import com.yuanshenbin.basic.util.NetworkUtils
import com.yuanshenbin.network.ResponseEnum
import com.yuanshenbin.network.model.ResponseModel
import java.io.Serializable

/**
 * author : yuanshenbin
 * time   : 2018/10/24
 * desc   :
 */
abstract class IDelegate : Serializable {
    open fun onShowToast(context: Context?, str: String?) {}
    open fun onStateLayout(model: ResponseModel?, layoutManager: StateLayoutManager?) {
        if (layoutManager != null && model != null) {
            when (model.state) {
                ResponseEnum.开始 -> layoutManager.showLoading()
                ResponseEnum.成功 -> layoutManager.showContent()
                ResponseEnum.结束 -> {
                }
                ResponseEnum.失败 -> if (NetworkUtils.isNetworkConnected(layoutManager.context)) {
                    layoutManager?.getNetworkState()?.showNetwork("网络似乎开小差了", "重试")
                } else {
                    layoutManager.showNetwork()


                }
                else -> {
                }
            }
        }
    }

    open fun onListDataEmpty(pullAndPush: Boolean, result: Any?): Boolean {
        return false
    }

    open fun handleListData(adapter: CommonAdapter<*>?, list: List<*>?, result: Any?, isPullAndPush: Boolean, loading: Int, page: Int, size: Int, emptyMsg: CharSequence?) {}
    open fun onTitle(title: String?) {}
    open fun onStart() {}
    open fun onStop() {}
    open fun onResume() {}
    open fun onPause() {}
    open fun onReload() {}
}